packages/boot/test/bootstrapTests/ec-contract-upgrade.test.ts

```diff
@@ -0,0 +1,645 @@
+  slotToBoardRemote,
+  unmarshalFromVstorage,
+} from '@agoric/vats/tools/board-utils.js';
+import { makeMarshal } from '@endo/marshal';
+
+import { makeSwingsetTestKit } from '../../tools/supports.js';
+import { makeWalletFactoryDriver } from '../../tools/drivers.js';
+
+const wallets = {
+  gov1: 'agoric1gx9uu7y6c90rqruhesae2t7c2vlw4uyyxlqxrx',
+  gov2: 'agoric1d4228cvelf8tj65f4h7n2td90sscavln2283h5',
+  gov3: 'agoric14543m33dr28x7qhwc558hzlj9szwhzwzpcmw6a',
+  gov4: 'agoric13p9adwk0na5npfq64g22l6xucvqdmu3xqe70wq',
+  gov5: 'agoric1el6zqs8ggctj5vwyukyk4fh50wcpdpwgugd5l5',
+  gov6: 'agoric1zayxg4e9vd0es9c9jlpt36qtth255txjp6a8yc',
+};
+
+const latestQuestionKey = `published.committees.Economic_Committee.latestQuestion`;
+const lastOutcomeKey = `published.committees.Economic_Committee.latestOutcome`;
+const highPrioritySenderKey = 'highPrioritySenders';
+
+const offerIds = {
+  propose: { outgoing: 'outgoing_propose' },
+  vote: { outgoing: 'outgoing_vote', incoming: 'incoming_vote' },
+};
+
+const getQuestionId = id => `propose-question-${id}`;
+const getVoteId = id => `vote-${id}`;
+
+export const makeZoeTestContext = async t => {
+  console.time('ZoeTestContext');
+  const swingsetTestKit = await makeSwingsetTestKit(t.log, undefined, {
+    configSpecifier: '@agoric/vm-config/decentral-main-vaults-config.json',
+  });
+
+  const { runUtils, storage } = swingsetTestKit;
+  console.timeLog('DefaultTestContext', 'swingsetTestKit');
+  const { EV } = runUtils;
+
+  await eventLoopIteration();
+
+  // We don't need vaults, but this gets the brand, which is checked somewhere
+  // Wait for ATOM to make it into agoricNames
+  await EV.vat('bootstrap').consumeItem('vaultFactoryKit');
+  console.timeLog('DefaultTestContext', 'vaultFactoryKit');
+
+  // has to be late enough for agoricNames data to have been published
+  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(storage);
+  console.timeLog('DefaultTestContext', 'agoricNamesRemotes');
+
+  console.timeEnd('DefaultTestContext');
+  const walletFactoryDriver = await makeWalletFactoryDriver(
+    runUtils,
+    storage,
+    agoricNamesRemotes,
+  );
+
+  const smartWallets = await Promise.all(
+    Object.values(wallets).map(async addr =>
+      walletFactoryDriver.provideSmartWallet(addr),
+    ),
+  );
+  const { fromCapData } = makeMarshal(undefined, slotToBoardRemote);
+
+  const getUpdatedDebtLimit = () => {
+    const atomGovernance = unmarshalFromVstorage(
+      storage.data,
+      'published.vaultFactory.managers.manager0.governance',
+      fromCapData,
+      -1,
+    );
+    return atomGovernance.current.DebtLimit.value.value;
+  };
+  return {
+    ...swingsetTestKit,
+    storage,
+    smartWallets,
+    fromCapData,
+    getUpdatedDebtLimit,
+  };
+};
+const test = anyTest as TestFn<Awaited<ReturnType<typeof makeZoeTestContext>>>;
+
+test.before(async t => {
+  t.context = await makeZoeTestContext(t);
+});
+
+test.serial('normal running of committee', async t => {
```

Comment:
#10163 uses the `governanceDriver` in `bootstrapTests`. It vastly simplifies the setup and interactions in order to propose and vote on parameter changes. Calling `updateVaultManagerParams()` does all the work for param changes.

The driver also has an entrypoint for accepting invitations. I don't know whether you'd have to enhance it to support changing electorates.

Reply:
I wondered about that driver... I sort of assumed it was limited to puppet governance, but I guess not?

Reply:
didn't appear to be. `boot/tools/drivers.ts` is definitely enumerating ec members during voting.

Reply:
Those are fine changes. I really like the way it simplified the test.